### PR TITLE
feat(ModelArts): add a new resource to manage service

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccModelartsService_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts -v

=== RUN   TestAccModelartsService_basic
=== PAUSE TestAccModelartsService_basic
=== CONT  TestAccModelartsService_basic
--- PASS: TestAccModelartsService_basic (512.26s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts	512.311s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_service_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    time_unit = "HOURS"
+  }
+}
+`, name, name)
+}
+
+func testModelartsService_basic_update(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_smn_topic" "test" {
+  name         = "%s"
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_service_test.go]: %[1]s maybe more clear

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go [hunk 2]
```diff
@@ -0,0 +1,1007 @@
...
+	if err != nil {
+		return diag.Errorf("error waiting for the Create of ModelartsService (%s) to complete: %s", d.Id(), err)
+	}
+	return resourceModelartsServiceRead(ctx, d, meta)
+}
+
+func buildCreateServiceBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"workspace_id":          utils.ValueIngoreEmpty(d.Get("workspace_id")),
+		"service_name":          utils.ValueIngoreEmpty(d.Get("name")),
```
Comment (Lance52259) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go]: the name, infer_type, config are required parameters, they would never be empty.

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_service_test.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_service_test.go]: ok

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go]: done.

Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go]: but It's better to keep it independent from the schema. 
If a field in the schema is changed to be optional, we just need to change the schema.

Comment (Lance52259) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go]: that will be the repsonsibility for update PR should have

File: docs/resources/modelarts_service.md [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    + **real-time**: A real-time service. A model is deployed as a web service and provides real-time test UI and monitoring.
+    + **batch**: A batch service, which can perform inference on batch data and automatically stops after data is processed.
+    + **edge**: An edge service, which uses Intelligent EdgeFabric (IEF) to deploy a model as a web service on an edge
+                node created on IEF.
+
+  Changing this parameter will create a new resource.
+
+* `config` - (Required, List) Model running configurations.  
+  If **infer_type** is **batch** or **edge**, you can configure only one model.
+  If**infer_type**is**real-time**, you can configure multiple models and assign weights based on service requirements.
```
Comment (ShiChangkuo) [inline docs/resources/modelarts_service.md]: please add spaces.

File: docs/resources/modelarts_service.md [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+  By default, this parameter is left blank.
+  This parameter is mandatory if vpc_id is configured.
+  A security group is a virtual firewall that provides secure network access control policies for service instances.
+  A security group must contain at least one inbound rule to permit the requests whose protocol is TCP,
+  source address is 0.0.0.0/0, and port number is 8080.
+
+  Changing this parameter will create a new resource.
+
+* `schedule` - (Optional, List) Service scheduling configuration, which can be configured only for real-time services.  
+  By default, this parameter is not used. Services run for a long time.
```
Comment (ShiChangkuo) [inline docs/resources/modelarts_service.md]: how should I understand about `this parameter is not used`?

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_service_test.go [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+resource "huaweicloud_modelarts_service" "test" {
+  name        = "%s"
+  description = "This is a acc test"
+  infer_type  = "real-time"
+
+  config {
+    specification  = "modelarts.vm.gpu.p4u8.container"
+    instance_count = 1
+    weight         = 100
+    model_id       = "48a44dd4-b15d-48b7-989a-f5b5f2d9a659"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_service_test.go]: please replace the UUID after #3171 is merged.

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go [hunk 6]
```diff
@@ -0,0 +1,1007 @@
...
+				Computed: true,
+			},
+			"additional_properties": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Elem:     modelartsServiceAdditionalPropertySchema(),
+				Optional: true,
+				Computed: true,
+			},
+			"change_status_to": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go:121]: why not use `status` directly?

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go [hunk 7]
```diff
@@ -0,0 +1,1007 @@
...
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Online debugging address of a real-time service.`,
+			},
+			"is_free": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: `Whether a free-of-charge flavor is used.`,
+			},
+			"publish_at": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go]: the following four arguments are hard-understanding. suggest to convert them to UTC format or delete them.

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go [hunk 8]
```diff
@@ -0,0 +1,1007 @@
...
+}
+
+func serviceWaitingForStateCompleted(ctx context.Context, d *schema.ResourceData, meta interface{}, t time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			cfg := meta.(*config.Config)
+			region := cfg.GetRegion(d)
+			// createServiceWaiting: missing operation notes
```
Comment (ShiChangkuo) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go]: please delete this line

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go:121]: The service may be stopped after running for some hours if set `schedule`.
So a separate field is better.

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go (hunk 7 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go]: deleted

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go (hunk 8 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go]: done

File: docs/resources/modelarts_service.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/modelarts_service.md]: done

File: docs/resources/modelarts_service.md (hunk 4 above)
Comment (hmbig2) [reply docs/resources/modelarts_service.md]: fixed

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_service_test.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_service_test.go]: ok

**Pull request merged**